### PR TITLE
253337 Integrate Managed Identity to Web project

### PR DESCRIPTION
**Pull request by gldefra**

### CODE REVIEW

File: EST.MIT.Web/Pages/create-invoice/OrganisationMetaSelection/OrganisationMetaSelection.razor.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ protected override async Task OnInitializedAsync()
         await base.OnInitializedAsync();
         invoice = _invoiceStateContainer.Value;
 
-        if (!invoice.IsNull())
+        if (invoice != null)
```
Comment (andygibbonsdefra) [inline EST.MIT.Web/Pages/create-invoice/OrganisationMetaSelection/OrganisationMetaSelection.razor.cs:28]: Would probably use "invoice is not null" over "invoice != null". Not important for this PR.

File: EST.MIT.Web/Services/ApprovalService.cs [hunk 2]
```diff
@@ -1,30 +1,19 @@
...
-    Task<Dictionary<string, string>> GetApproversAsync(string scheme, string value);
-    Task<Invoice> GetApprovalAsync(string id, string scheme);
-    Task<IEnumerable<Invoice>> GetOutstandingApprovalsAsync();
-    Task<ApiResponse<BoolRef>> ValidateApproverAsync(string approver);
-}
-
 public class ApprovalService : IApprovalService
 {
-    private readonly IQueueService _queueService;
+    private readonly IEventQueueService _queueService;
```
Comment (andygibbonsdefra) [inline EST.MIT.Web/Services/ApprovalService.cs:10]: Probably would have named this _eventQueueService too, for clarity, if there was renaming going on.

File: EST.MIT.Web/Collections/AzureServices.cs [hunk 3]
```diff
@@ -1,15 +1,55 @@
...
 namespace EST.MIT.Web.Collections;
-
 public static partial class ServiceCollectionExtensions
 {
-    public static IServiceCollection AddAzureServices(this IServiceCollection services)
+    public static IServiceCollection AddAzureServices(this IServiceCollection services, IConfiguration configuration)
     {
-        services.AddSingleton<IBlobService, BlobService>();
-        services.AddSingleton<IQueueService, QueueService>();
         services.AddSingleton<IAzureBlobService, AzureBlobService>();
```
Comment (jbarnsley10) [inline EST.MIT.Web/Collections/AzureServices.cs:12]: Remove this line. You're effectively defining the Singleton twice

Comment (gldefra) [reply EST.MIT.Web/Collections/AzureServices.cs:12]: good spot 👍

File: EST.MIT.Web/Pages/create-invoice/OrganisationMetaSelection/OrganisationMetaSelection.razor.cs (hunk 1 above)
Comment (gldefra) [reply EST.MIT.Web/Pages/create-invoice/OrganisationMetaSelection/OrganisationMetaSelection.razor.cs:28]: we do need to look at Sonar cloud as it is flagging **(!invoice.IsNull())** as a code smell, hence the change

Comment (andygibbonsdefra) [reply EST.MIT.Web/Pages/create-invoice/OrganisationMetaSelection/OrganisationMetaSelection.razor.cs:28]: No worries.

**Review by andygibbonsdefra — APPROVED** [review]

**Pull request merged**